### PR TITLE
[#566] ansible packaging

### LINKED ISSUE #566: Create CI build pipeline for cluster infrastructure components to produce set of deliverable infrastructure units
Create Jenkins build pipeline for cluster infrastructure components. 
This pipeline should include:
1) build of 3d party customised docker containers.
     - move infra related dockerfiles (kube-fluentd, elb-security, ?) to k8s/k8s-infra
     - create jenkins job for Infra images build
     - update release pipeline with new step for infra 
     - update playbooks\charts\etc with new images location
     - update existing jenkins jobs with new infra version tag parameter
     - include dex build into this build pipeline. We customised dex in https://github.com/legion-platform/dex/commits/feat/oidc-groups. Rename branch  oidc-groups - > legion and set legion release tag.
2) package helm charts (to be discussed )
     - public repo for releases https://github.com/helm/charts
     - private repo for snapshots https://github.com/sonatype-nexus-community/nexus-repository-helm 
3) package ansible scripts docker image
     - public repo for releases docker hub
     - private repo for snapshots nexus

Single version of Legion infrastructure should be set for all packages and containers listed above. Packages and container should be published to repositories (Nexus or public repos - to be discussed)

Create cluster pipeline should depend on these containers and packages but not on Legion git repository.

### PR DESCRIPTION
**Pull request by alexander-semenets**
it closes #566

### CODE REVIEW

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

File: deploy/ci-infra.Jenkinsfile [hunk 1]
```diff
@@ -6,10 +6,62 @@ node {
...
+            cat "${logFile.getPath()}" | perl -pe 's/\\x1b\\[8m.*?\\x1b\\[0m//g;' > build-log.txt 2>&1
+            """
+            archiveArtifacts 'build-log.txt'
+
+            // Copy artifacts
+            copyArtifacts filter: '*', flatten: true, fingerprintArtifacts: true, projectName: 'Build_Legion_Artifacts', selector: specific(buildNumber.toString()), target: ''
+            sh 'ls -lah'
+            
+            //sh 'cp pylint.log python-lint-log.txt'
+            //archiveArtifacts 'python-lint-log.txt'
```
Comment (dsuslov) [inline deploy/ci-infra.Jenkinsfile]: @alexander-semenets why is it commented out?

File: deploy/ci-infra.Jenkinsfile [hunk 2]
```diff
@@ -6,10 +6,62 @@ node {
...
+
+            // Save logs
+            logFile = result.getRawBuild().getLogFile()
+            sh """
+            cat "${logFile.getPath()}" | perl -pe 's/\\x1b\\[8m.*?\\x1b\\[0m//g;' > build-log.txt 2>&1
+            """
+            archiveArtifacts 'build-log.txt'
+
+            // Copy artifacts
+            copyArtifacts filter: '*', flatten: true, fingerprintArtifacts: true, projectName: 'Build_Legion_Artifacts', selector: specific(buildNumber.toString()), target: ''
```
Comment (dsuslov) [inline deploy/ci-infra.Jenkinsfile]: @alexander-semenets Why is the target not defined? What is default value of target?

File: k8s/ansible/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,53 @@
+FROM ubuntu:18.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+# install system utilities
+RUN apt-get update && apt-get install -y \
```
Comment (dsuslov) [inline k8s/ansible/Dockerfile]: @alexander-semenets pin versions of 3d party tools

Comment (alexander-semenets) [reply k8s/ansible/Dockerfile]: this is pinned at Ubuntu revision level, there are only minor and security updates usually pushed to per release (16.04, 18.04, etc) repositories

File: deploy/ci-infra.Jenkinsfile (hunk 2 above)
Comment (alexander-semenets) [reply deploy/ci-infra.Jenkinsfile]: will be reviewed and cleaned up in #649 story

File: deploy/ci-infra.Jenkinsfile (hunk 1 above)
Comment (alexander-semenets) [reply deploy/ci-infra.Jenkinsfile]: will be reviewed and cleaned up in #649 story

**Pull request merged**